packages/meditrak-server/src/apiV2/export/exportSurveyResponses/exportResponsesToFile.js

```diff
@@ -0,0 +1,306 @@
+      surveyResponseIds,
+      {},
+      { columns: [{ [`${TYPES.SURVEY_RESPONSE}.id`]: 'survey_response.id' }], sort: [] },
+    );
+
+    // Add any questions that are in survey responses but no longer in the survey
+    const allQuestionIds = getUniqueEntries(answers.map(a => a['question.id']).flat());
+    const validQuestionIds = questions.map(q => q.id);
+    const outdatedQuestionIds = allQuestionIds.filter(id => !validQuestionIds.includes(id));
+    const outdatedQuestions = await models.question.find({ id: outdatedQuestionIds });
+    const allQuestions = [...questions, ...outdatedQuestions];
+
+    // Add title
+    if (reportName) {
+      exportData.unshift([`${reportName} - ${survey.name}, ${country.name}`]);
+    }
+
+    // Exclude 'SubmissionDate'/'DateOfData' and 'PrimaryEntity' rows from survey response export since these have no answers
+    const questionsForExport = allQuestions.filter(
+      ({ type: questionType }) =>
+        !NON_DATA_ELEMENT_ANSWER_TYPES.includes(questionType) ||
+        questionType === ANSWER_TYPES.INSTRUCTION,
+    );
+
+    // Add the questions info and answers to be exported
+    const preQuestionRowCount = exportData.length;
+
+    // Set up the left columns with info about the questions
+    questionsForExport.forEach((question, questionIndex) => {
+      const questionInfo = infoColumnKeys.map(columnKey => question[columnKey]);
+      const exportRow = preQuestionRowCount + questionIndex;
+      exportData[exportRow] = questionInfo;
+    });
+
+    // Add the answers on the right columns to the exportData
+    const questionIdToIndex = Object.fromEntries(
+      questionsForExport.map((question, index) => [question.id, index]),
+    );
+    const responseIdToIndex = Object.fromEntries(
+      surveyResponses.map((response, index) => [response.id, index]),
+    );
+    answers.forEach(answer => {
+      const surveyResponseIndex = responseIdToIndex[answer['survey_response.id']];
+      const questionIndex = questionIdToIndex[answer['question.id']];
+      const exportRow = preQuestionRowCount + questionIndex;
+      const exportColumn = infoColumnKeys.length + surveyResponseIndex;
+      exportData[exportRow][exportColumn] = answer?.text || '';
+    });
+
+    if (easyReadingMode) {
+      return addExportedDateAndOriginAtTheSheetBottom(exportData, timeZone);
+    }
+
+    return exportData;
+  };
+
+  const addResponsesToSheet = async (surveyResponses, survey, questions) => {
+    const exportData = await getExportDataForResponses(surveyResponses, survey, questions);
+    addDataToSheet(survey.name, exportData);
+  };
+
+  /** Main body of the function below this point, everything above is helper functions */
+
+  const surveysWithAccessStatus = await Promise.all(
+    surveys.map(async s => ({ ...s, accessible: await checkAccessToSurvey(s) })),
+  );
+  const surveysByAccessStatus = groupBy(surveysWithAccessStatus, 'accessible');
+  surveysByAccessStatus[false]?.forEach(survey => {
```

Comment:
```suggestion
  (surveysByAccessStatus[false] || []).forEach(survey => {
```

Although it would be event better to extract those:
```js
  const { true: surveysWithAccess = [], false: surveysWithoutAccess = [] } = surveysByAccessStatus;
  // etc
```

Reply:
The `?` deals with it: "only run the thing after the `.` if there's something defined before the `?`"

I do think your syntax is a good suggestion though, will make that change

Reply:
Oh I see - so it's not like we'll try to run an `undefined` function. That's convenient!